mlflow/pyfunc/__init__.py

```diff
@@ -1810,10 +1993,37 @@ def spark_udf(
         # this case all executors and driver share the same filesystem
         is_spark_in_local_mode = spark.conf.get("spark.master").startswith("local")
 
+    dbconnect_mode = is_databricks_connect(spark)
+    if prebuilt_env_path is not None and not dbconnect_mode:
+        raise RuntimeError(
+            "'prebuilt_env' param can only be used in Databricks Serverless notebook REPL, "
+            "Databricks Shared cluster notebook REPL, and Databricks Connect client "
+            "environment."
+        )
+
+    if (
+        prebuilt_env_path is None
+        and env_manager != _EnvManager.LOCAL
+        and dbconnect_mode
```

Comment:
Is `dbconnect_mode` a string? or a boolean? `dbconnect_mode` sounds like it's a string.